crates/red_knot_python_semantic/src/types.rs

```diff
@@ -874,6 +874,9 @@ impl<'db> Type<'db> {
                     function_type.return_type(db),
                     *arg_types.first().unwrap_or(&Type::Unknown),
                 ),
+                Some(KnownFunction::IsInstance) => {
+                    CallOutcome::callable(KnownClass::Bool.to_instance(db))
+                }
```

Comment:
Given that `IsInstance` doesn't have special behavior when called in type inference, I think it would be better to defer to typeshed rather than hardcoding its return type. So I would use `function_type.return_type(db)` here, or even better collapse this with the `None` arm.

Reply:
With Alex's suggestion above of a `FunctionType::is_known` method, this could probably just become an `if function_type.is_known(KnownFunction::RevealType)` instead of a match, at least until we have another known function with special inference-time behavior.